### ISSUE TITLE
Support for routing based on path and method?

### ISSUE THREAD

**Issue by memelet** [eBay/fabio#247]
Is there, or could there be, support routing based on a combination of the path and method?

As we migrate away from some legacy systems we need to just a little bit of routing magic. Right now we use haproxy with consul-template to build the config.

**Comment by magiconair** [comment]
Not yet but I could add that, e.g. `urlprefix-/foo method=GET,POST` ?

**Comment by memelet** [comment]
That would work just great.

**Comment by magiconair** [comment]
Yeah, I thought that would be a simple change and started working on that but that rabbit hole is a bit deeper than I thought. Need to give this a bit more thought.

**Comment by memelet** [comment]
Ok, no hurry. Thanks!

**Comment by ket4yii** [comment]
What about to make this feature is able to support protocols? Because a lot of users, for example, make requests to http when you've got https server and their requests will fail. I know, that there are many ways, to do it, like make another tls terminator, proxy and so on, but it would be really convenient if you implement this.

**Comment by magiconair** [comment]
@ket4yii I think that is a different feature you're asking for which is `redirect a -> b`

**Comment by ket4yii** [comment]
@magiconair Yeah, maybe it's the better way. Is there any way to do it?

**Comment by magiconair** [comment]
Not right now but you haven't been the first to ask. I'll add it at some point. Sorry, if I can't be more specific.

**Comment by deuch** [comment]
I've the same use case with cqrs micro-service. We would like to have the ability to have the same service/path and route request to the right backend regarding http method (GET for read and POST/PUT for write).

**Comment by magiconair** [comment]
OK, I'll have another look. Started with this at some point but it took more time than I expected.
